umh-core/pkg/fsm/redpanda_monitor/actions.go

```diff
@@ -0,0 +1,141 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package redpanda_monitor
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/constants"
+	"github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/logger"
+	"github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/metrics"
+	"github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/service/filesystem"
+	redpanda_monitor_service "github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/service/redpanda_monitor"
+	"github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/serviceregistry"
+)
+
+// CreateInstance is called when the FSM transitions from to_be_created -> creating.
+func (r *RedpandaMonitorInstance) CreateInstance(ctx context.Context, filesystemService filesystem.Service) error {
+	r.baseFSMInstance.GetLogger().Debugf("Starting Action: Adding Redpanda Monitor service %s to S6 manager ...", r.baseFSMInstance.GetID())
+
+	r.baseFSMInstance.GetLogger().Debugf("Adding Redpanda Monitor service %s to S6 manager", r.baseFSMInstance.GetID())
+	err := r.monitorService.AddRedpandaMonitorToS6Manager(ctx)
+	if err != nil {
+		if err == redpanda_monitor_service.ErrServiceAlreadyExists {
+			r.baseFSMInstance.GetLogger().Debugf("Redpanda Monitor service %s already exists in S6 manager", r.baseFSMInstance.GetID())
+			return nil // do not throw an error, as each action is expected to be idempotent
+		}
+		return fmt.Errorf("failed to add Redpanda Monitor service %s to S6 manager: %w", r.baseFSMInstance.GetID(), err)
+	}
+
+	r.baseFSMInstance.GetLogger().Debugf("Redpanda Monitor service %s added to S6 manager", r.baseFSMInstance.GetID())
+
+	return nil
+}
+
+// RemoveInstance is called when the FSM transitions to removing.
+// It requires the service to be stopped before removal.
+func (r *RedpandaMonitorInstance) RemoveInstance(ctx context.Context, filesystemService filesystem.Service) error {
+	r.baseFSMInstance.GetLogger().Debugf("Starting Action: Removing Redpanda Monitor service %s from S6 manager ...", r.baseFSMInstance.GetID())
+
+	// Remove the Redpanda from the S6 manager
+	err := r.monitorService.RemoveRedpandaMonitorFromS6Manager(ctx)
+	if err != nil {
+		if err == redpanda_monitor_service.ErrServiceNotExist {
+			r.baseFSMInstance.GetLogger().Debugf("Redpanda Monitor service %s not found in S6 manager", r.baseFSMInstance.GetID())
+			return nil // do not throw an error, as each action is expected to be idempotent
+		}
+		return fmt.Errorf("failed to remove Redpanda Monitor service %s from S6 manager: %w", r.baseFSMInstance.GetID(), err)
+	}
+
```

Comment:
I've added in the removal PR a big change here, the redpanda_monitor needs to have it as well

Reply:
As far as i can see, i already applied those.

Reply:
![image](https://github.com/user-attachments/assets/3358a000-1060-44f0-9c1d-f7aafb593649)compare it line by line with benthos_monitor, then you will see that this here is for example missing